geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaCacheLoader.scala

```diff
@@ -37,30 +36,37 @@ trait KafkaCacheLoader extends Closeable with LazyLogging {
     new ConcurrentHashMap[(SimpleFeatureSource, FeatureListener), java.lang.Boolean]()
   }
 
+  @volatile
+  private var hasListeners = false
```

Comment:
Any particular reason for preferring a variable over checking if the listeners.isEmpty?

Reply:
yeah, the implementation of isEmpty in ConcurrentHashMap is slightly expensive, so I thought it would be better to just set a flag. Normally, the listeners wouldn't change much (if ever). I'm adding some comments to explain

Here's the source for ConcurrentHashMap:

```
    public boolean isEmpty() {
        return sumCount() <= 0L; // ignore transient negative values
    }

    final long sumCount() {
        CounterCell[] as = counterCells; CounterCell a;
        long sum = baseCount;
        if (as != null) {
            for (int i = 0; i < as.length; ++i) {
                if ((a = as[i]) != null)
                    sum += a.value;
            }
        }
        return sum;
    }
```